src/python/pants/engine/internals/graph.py

```diff
@@ -127,17 +127,14 @@ async def resolve_target(
         return WrappedTarget(target)
 
     wrapped_generator_tgt = await Get(
-        WrappedTarget, Address, address.maybe_convert_to_build_target()
+        WrappedTarget, Address, address.maybe_convert_to_target_generator()
     )
     generator_tgt = wrapped_generator_tgt.target
-    # Not all languages support file targets (Go and Terraform). If we got a file address, use
-    # that as an alias for the BUILD target.
-    #
-    # This code path will go away once we remove file addresses in favor of generated target
-    # addresses, e.g. path/to:generator:generated, because we will error in that case when
-    # there is no generator for the address.
     if not target_types_to_generate_requests.is_generator(generator_tgt):
+        # TODO: Error in this case. You should not use a generator address (or file address) if
+        #  the generator does not actually generate.
```

Comment:
This seems like something that we could safely do now, since none exist at this point?

Reply:
Yep! I'm saving for a follow up to keep this smaller. 

Note that this will result in erroring when using file address syntax on Go and Terraform, which I think is the right behavior. And safe because they're experimental.

Reply:
Mm. But the CLI should still work fine when executed on a file, because it finds owning targets?

Reply:
Exactly. This only will error when you use `src/go/foo.go:pkg` and `src/go:pkg#foo.go`.